src/lib/core/fiber.c

```diff
@@ -2110,6 +2128,8 @@ cord_cojoin(struct cord *cord)
 		 */
 		do {
 			assert(cord->id != 0);
+			if (fiber_is_cancelled())
+				ev_async_send(cord->loop, &cord->cancel_event);
```

Comment:
This looks dubious. I don't think it's a good idea to stop a thread upon cancelling a fiber trying to join it. IMO it looks unexpected and may obscure the code. We need to discuss that.

Reply:
Discussed f2f and decided to keep it. Good for default and we can add option to change default behavior in the future.